src/app/dao/tests/JoinerDAO.spec.ts

```diff
@@ -1,17 +1,462 @@
+            const candidate: MinimalUser = await createConnectedUser(CANDIDATE_EMAIL, CANDIDATE_NAME);
+            await expectAsync(joinerDAO.addCandidate(partId, candidate)).toBeResolvedTo();
+
+            // When the user adds themself to the list of candidates a second time
+            // (for example, because they closed their tab and open it again)
+            const result: Promise<void> = joinerDAO.addCandidate(partId, candidate);
+
+            // Then it should succeed
+            // And there should be only one candidate document for the user
+            await expectAsync(result).toBeResolvedTo();
+            const matchingDocs: unknown[] = await joinerDAO.subCollectionDAO(partId, 'candidates').findWhere([['id', '==', candidate.id]]);
+            expect(matchingDocs.length).toBe(1);
+        });
+        it('should forbid to remove someone else from the candidates', async() => {
+            // Given a part and joiner, with another user as candidate, and a malicious user
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            const candidate: MinimalUser = await addCandidate(partId);
+            await createConnectedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+
+            // When the malicious users tries to remove another candidate
+            const result: Promise<void> = joinerDAO.removeCandidate(partId, candidate);
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+        it('should forbid changing the fields of another candidate', async() => {
+            // Given a part and joiner, with another user as candidate, and a malicious user
+            const partId: string = (await createPart({ createJoiner: true, signOut: true })).partId;
+            const candidate: MinimalUser = await addCandidate(partId);
+
+            await createConnectedUser(MALICIOUS_EMAIL, MALICIOUS_NAME);
+
+            // When the malicious users tries to change another candidate's fields
+            const update: Partial<MinimalUser> = { name: 'foo' };
+            const result: Promise<void> = joinerDAO.subCollectionDAO(partId, 'candidates').update(candidate.id, update);
+
+            // Then it should fail
+            await expectPermissionToBeDenied(result);
+        });
+    });
+    describe('for creator', () => {
+        it('should forbid setting a fake username as a creator', async() => {
+            // Given a verified user and an existing part
+            const createdPart: CreatedPart = await createPart();
+
+            // When creating the corresponding joiner, with the user as creator but with a fake username
+            const fakeCreator: MinimalUser = { id: createdPart.creator.id, name: 'fake-jeanjaja' };
+            const result: Promise<void> = joinerDAO.set(createdPart.partId, { ...JoinerMocks.INITIAL, fakeCreator });
```

Comment:
fakeCreator n'estp as un champs d'un doc joiner!
Ici il se plaint car tu n'as pas overridé creator et qu'c'pas toa !

Reply:
LE TYPAGE EST FAIBLE ! "you're a fucking disappointment" (moi, au linter :D ) (ps: tu aurais cette référence si tu avais terminé The Boys !!!!!)

Reply:
ouiii le typage de nos objets DAO est craiiiignosss je suis 100% akoord! C'est car ils sont des JSONObject donc ils peuvent avoir des champs supplémentaires.
Fixed

Reply:
ça mérite un bon ticket du super-linterisme ?
Genre les champs qu'on passe dans un DAO seraient plus des JSONObject mais des { type: Type<LaClasse>, data: JSONObject } et la première ligne de chaque méthode du CRUD serait de comparer type.maFonctionStatiqueQuiDitEnDurLesFieldsLegaux avec Object.keys(update) pour voir si y a des trucs à gauche qui sont pas à droite (l'inverse étant accepté, mais pas ça) :D 

Ou juste un typage plus ... typé :D

Reply:
Ajouté un ticket car il faut un typage plus typé clairement :D